ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracingshim/__init__.py

```diff
@@ -0,0 +1,257 @@
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class TracerWrapper(opentracing.Tracer):
+    def __init__(self, tracer, scope_manager=None):
+        # If a scope manager isn't provided by the user, create a
+        # `ScopeManagerWrapper` instance and use it to initialize the
+        # `TracerWrapper`.
+        if scope_manager is None:
+            scope_manager = ScopeManagerWrapper(self)
+        super().__init__(scope_manager=scope_manager)
+        self._otel_tracer = tracer
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Tracer` object."""
+
+        return self._otel_tracer
+
+    @contextmanager
+    def start_active_span(
+        self,
+        operation_name,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+        finish_on_close=True,
+    ):
+        span = self.start_span(
+            operation_name=operation_name,
+            child_of=child_of,
+            references=references,
+            tags=tags,
+            start_time=start_time,
+            ignore_active_span=ignore_active_span,
+        )
+        with self._scope_manager.activate(span, finish_on_close) as scope:
+            yield scope
+
+    def start_span(
+        self,
+        operation_name=None,
+        child_of=None,
+        references=None,
+        tags=None,
+        start_time=None,
+        ignore_active_span=False,
+    ):
+        if child_of is None:
+            parent = None
+        else:
+            # TODO: Should we use the OpenTracing base classes for the type
+            # check?
+            if isinstance(child_of, (SpanWrapper, SpanContextWrapper)):
+                # The parent specified in `child_of` is valid and is either a
+                # `SpanWrapper` or a `SpanContextWrapper`. Unwrap the `Span` or
+                # `SpanContext` to extract the OpenTracing object and use this
+                # object as the parent of the created span.
+                parent = child_of.unwrap()
+            else:
+                logger.warning(
+                    "Unknown class %s passed in child_of argument to start_span() method.",
+                    type(child_of),
+                )
+                parent = None
+                # TODO: Refuse to create a span and return `None` instead of
+                # proceeding with a `None` parent? This would cause the created
+                # span to become a child of the active span, if any, or create
+                # a new trace and make the span the root span of that trace.
+
+        # Use active span as parent when no explicit parent is specified.
+        if (
+            self.active_span is not None
+            and not ignore_active_span
+            and not parent
+        ):
+            parent = self.active_span.unwrap()
+
+        span = self._otel_tracer.create_span(operation_name, parent)
```

Comment:
Please use `start_span()` as I plan to have a PR remove `create_span()` ;)

Reply:
Looks like I can't at the moment - `start_span()` isn't flexible enough AFAICT:

- We need to support the `start_time` argument as expected by the OT API. We would need to add a `start_time` argument to `start_span()` and use it in the SDK and other implementations when calling `span.start()`.
- `start_span()` calls `use_span()` which returns a context manager. This means we cannot get a `Span` object without triggering `__enter__()` on the context manager, which starts the span. This is a problem because we need to call methods on `Span` **before** starting the span: https://github.com/kinvolk/opentelemetry-python/blob/78e58a557c739f9b3358b04034189dab8c69ca75/ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py#L254-L262

I can look into making `start_span()` more flexible if this makes sense @carlosalberto.

Reply:
`start_span()` needs to support an start timestamp anyway (eventually).

Also, I hope I can merge the PR to **not** have `start_span()` as context manager, but as a `Span`. My bad on being slow there ;(

(Also, IHMO it's not so important to support explicit timestamps in the OT shim at this very moment - we can throw an exception for now).